store.js

```diff
@@ -0,0 +1,358 @@
+    cartRow.classList.add('cart-row')
+    const cartItems = document.getElementsByClassName('cart-items')[0]
+    const itemTitle = document.getElementsByClassName('shop-item-title')
+    const cartItemName = cartItems.getElementsByClassName('cart-item-title')
+    Array.from(cartItemName).map((el, index) => {
+        if (el.textContent === title) {
+            let quantityElement = cartItems.getElementsByClassName('cart-quantity-input')[index]
+            quantityElement.value = parseInt(quantityElement.value) + 1
+        }
+    })
+    const cartRowContent = `
+        <div class="cart-item cart-column">
+            <span class="cart-item-title">${title}</span>
+            
+        </div>
+        <span class="cart-price cart-column">${price}</span>
+        <div class="cart-quantity cart-column">
+            <input class="cart-quantity-input" type="number" value="${count}">
+            <button class="btn btn-removeItemCard" type="button">Usuń</button>
+        </div>`
+    cartRow.innerHTML = cartRowContent
+    cartItems.append(cartRow)
+    cartRow.getElementsByClassName('btn-removeItemCard')[0].addEventListener('click', removeCartItem)
+    cartRow.getElementsByClassName('cart-quantity-input')[0].addEventListener('change', quantityChanged)
+    if (cartItemName.length >= 1) total()
+}
+
+const updateCartTotal = () => {
+    const basket = getLocalStorage('basket')
+    let total = 0
+    Array.from(basket).map((el) => {
+        let price = el.price
+        price = price.replace(' zł', '');
+        let quantity = el.count
+        total = total + (price * quantity)
+    })
+    total = Math.round(total * 100) / 100;
+    document.getElementsByClassName('cart-total-price')[0].textContent = total + ' zł'
+}
+
+const total = () => {
+    const cartTotal = document.getElementsByClassName('cart-total-title')[0]
+    const cartRowContent = "Do zapłaty:"
+    cartTotal.textContent = cartRowContent
+}
+const totalBack = () => {
+    const cartTotal = document.getElementsByClassName('cart-total-title')[0]
+    const cartRowContent = "Głodny? Zamów pizze!"
+    cartTotal.textContent = cartRowContent
+    document.getElementsByClassName('cart-total-price')[0].textContent = ""
+
+}
+
+const showShopItem = async () => {
+    const url = 'https://raw.githubusercontent.com/alexsimkovich/patronage/main/api/data.json'
+    let response = await fetch(url)
+    let data = await response.json()
+    return data
+}
+
+const filtr = () => {
+    const input = document.getElementById("filtr")
+    let filter = input.value
+    const cardContainer = document.getElementById("shop-items")
+    const cards = cardContainer.getElementsByClassName("shop-item")
+    Array.from(cards).map((el) => {
+        const ingredients = el.querySelector(".shop-item-ingredients")
+        if (ingredients.textContent.indexOf(filter) > -1) {
+            el.style.display = ""
+        } else {
+            el.style.display = "none"
+        }
+    })
+}

```

Comment:
Wskazówka: musisz mieć dostęp do danych (tablica pizz), które pobrałaś na początku. Filtrowanie musisz zrobić po tablice i na nowo wyrenderować HTML-a podająć tylko odfiltrowane dane. To samo dotyczy sortowania. Wszystkie manipulacje `filter, sort` trzeba robić na tablice z danymi, a nie na HTML-u.
Example:
```
const filter = (data) => {
  const filteredList = data.filter(item => {
    your code here ...
  })
  renderPizzasz(filteredList)
}
```